service/src/main/java/com/epam/rft/atsy/service/impl/PositionServiceImpl.java

```diff
@@ -82,6 +82,11 @@ public Long saveOrUpdate(PositionDTO position) {
       throw new DuplicatePositionException(position.getName());
     }
 
-    return this.positionRepository.saveAndFlush(positionEntity).getId();
+    try {
+      return this.positionRepository.saveAndFlush(positionEntity).getId();
+    } catch (NullPointerException e) {
+      return -1L;
```

Comment:
For me it seems like a black magic. Why does this method return -1 if null pointer exception is occurred? In my opinion it is more readable if throw illegal argument exception if a validation fails or return null if savedEntity is null, but I prefer my first option. And one more question: Could you please describe a situation when savedEntity can be null?